ilc/spec/wrapWithCache.spec.js

```diff
@@ -0,0 +1,177 @@
+import chai from 'chai';
+import sinon from 'sinon';
+import wrapWithCache from '../common/wrapWithCache';
+
+describe('wrapWithCache', () => {
+    let wrapedFn;
+    let wrapWithCacheStorage;
+
+    const cacheStorage = new Map();
+    const fn = sinon.stub();
+    const createHash = sinon.stub();
+    const logger = {
+        error: sinon.stub(),
+    };
+
+    const errorThrowedByFn = new Error('Error message');
+
+    before(() => {
+        wrapWithCacheStorage = wrapWithCache(cacheStorage, logger, createHash);
+    });
+
+    afterEach(() => {
+        fn.resetBehavior();
+        createHash.resetBehavior();
+        logger.error.resetBehavior();
+        cacheStorage.clear();
+    });
+
+    describe('when cache storage is empty', () => {
+        const data = 'data';
+        const cacheParams = {
+            cacheForSeconds: 60,
+        };
+
+        before(() => {
+            wrapedFn = wrapWithCacheStorage(fn, cacheParams);
+        });
+
+        beforeEach(() => {
+            fn.withArgs().returns(Promise.resolve(data));
+        });
+
+        it('should return a cached value when a user is calling a cached function', async () => {
+            const cachedValue = await wrapedFn();
+
+            chai.expect(cachedValue).to.own.include({
+                data,
+            }).and.to.have.all.keys('checkAfter', 'cachedAt');
+        });
+
+        it('should save a value into the cache', async () => {
+            await wrapedFn();
+
+            chai.expect(cacheStorage.size).to.equal(1);
```

Comment:
I don't think this is a correct test. Let's discuss

Reply:
Updated